File: services/typescript/hono/src/routes/handlers/invalid_path.ts
import type { NotFoundHandler } from "hono";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "../../logger.js";

export const invalidPathHandler: NotFoundHandler = (c) => {
  getLogger().warn({ method: c.req.method, path: c.req.path }, "Invalid path");
  return sendError(c, AppErrors.InvalidPath);
};

File: services/python/sanic/src/routes/handlers/invalid_path.py
import structlog
from sanic import Request
from sanic.response import JSONResponse
from sanic.exceptions import NotFound

from routes.errors import AppErrors, send_error


async def invalid_path_handler(request: Request, _exception: NotFound) -> JSONResponse:
    structlog.get_logger().warn(
        "Invalid path", method=request.method, path=request.path
    )

    return send_error(AppErrors.InvalidPath)
